### PR TITLE
Nonnegative predictions for deepar mxnet

### PR DESCRIPTION
**Pull request by melopeo**
*Issue #, if available:*

*Description of changes:*
Add functionality to generate nonnegative prediction samples for DeepAR. This is applied only to final samples for prediction.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: test/mx/model/deepar/test_nonnegative_pred_samples.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    dataset_train, dataset_test = datasets
+    predictor = estimator.train(dataset_train)
+    forecasts = list(predictor.predict(dataset_test))
+    assert all([forecast.samples.dtype == dtype for forecast in forecasts])
+    assert len(forecasts) == len(dataset_test)
+
+    if nonnegative_pred_samples:
+        assert all(
+            [(forecast.samples < 0).sum() == 0 for forecast in forecasts]
+        )
```
Comment (lostella) [inline test/mx/model/deepar/test_nonnegative_pred_samples.py]: Minor, but more readable I think

```suggestion
        for forecast in forecasts:
            assert (forecast.samples >= 0).all()
```

File: test/mx/model/deepar/test_nonnegative_pred_samples.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    datasets,
+    dtype,
+    impute_missing_values,
+    nonnegative_pred_samples,
+):
+    estimator = DeepAREstimator(
+        distr_output=distr_output,
+        dtype=dtype,
+        impute_missing_values=impute_missing_values,
+        nonnegative_pred_samples=nonnegative_pred_samples,
```
Comment (lostella) [inline test/mx/model/deepar/test_nonnegative_pred_samples.py]: Given the name of the test, maybe the option should be hard-coded?

```suggestion
def test_deepar_nonnegative_pred_samples(
    distr_output,
    datasets,
    dtype,
    impute_missing_values,
):
    estimator = DeepAREstimator(
        distr_output=distr_output,
        dtype=dtype,
        impute_missing_values=impute_missing_values,
        nonnegative_pred_samples=True,
```

File: src/gluonts/mx/model/deepar/_network.py [hunk 3]
```diff
@@ -789,6 +791,24 @@ def unroll_encoder_default(
...
+        ----------
+        samples
+            Tensor of shape (batch_size * num_samples, 1)
+        Returns
+        -------
+            Tensor of samples with the same shape.
+        """
+
+        if self.nonnegative_pred_samples:
+            return F.Activation(samples, act_type="relu")
```
Comment (lostella) [inline src/gluonts/mx/model/deepar/_network.py]: I'm wondering if this works both in symbolic vs eager mode

```suggestion
            return F.relu(samples)
```

File: test/mx/model/deepar/test_nonnegative_pred_samples.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        impute_missing_values=impute_missing_values,
+        nonnegative_pred_samples=True,
+        freq="D",
+        prediction_length=3,
+        trainer=Trainer(epochs=1, num_batches_per_epoch=1),
+    )
+
+    dataset_train, dataset_test = datasets
+    predictor = estimator.train(dataset_train)
+    forecasts = list(predictor.predict(dataset_test))
```
Comment (lostella) [inline test/mx/model/deepar/test_nonnegative_pred_samples.py:51]: To ensure the added feature also works in symbolic mode, you can call `as_symbol_block_predictor` on the predictor object, see https://github.com/awslabs/gluonts/blob/a818f69dc049c1c1d57e09d2ccb8b5f7a0cff656/src/gluonts/mx/model/predictor.py#L316

**Review by lostella — APPROVED** [review]
